prov/rxm/src/rxm.h

```diff
@@ -127,18 +131,22 @@ struct rxm_rma_iov {
 	struct ofi_rma_iov iov[];
 };
 
-/* States for large message transfer */
-#define RXM_LMT_STATES(FUNC)	\
-	FUNC(RXM_LMT_NONE),	\
-	FUNC(RXM_LMT_START),	\
-	FUNC(RXM_LMT_ACK),	\
+/* RXM protocol states / tx/rx context */
+#define RXM_PROTO_STATES(FUNC)	\
+	FUNC(RXM_NONE),		\
+	FUNC(RXM_TX),		\
+	FUNC(RXM_RX),		\
+	FUNC(RXM_LMT_TX),	\
+	FUNC(RXM_LMT_ACK_WAIT),	\
+	FUNC(RXM_LMT_READ),	\
+	FUNC(RXM_LMT_ACK_SENT), \
 	FUNC(RXM_LMT_FINISH),
 
-enum rxm_lmt_state {
-	RXM_LMT_STATES(ENUM)
+enum rxm_proto_state {
+	RXM_PROTO_STATES(ENUM)
```

Comment:
This macro is used to avoid duplication of code when defining rxm_proto_state_str (rxm.c).

RXM_PROTO_STATES accepts another macro as argument. In the above code where we are just defining the states, we use the macro ENUM which is defined as `#define ENUM(X) X` (rxm.h). When defining the corresponding strings (rxm.c) we pass the stringify macro, STR which is defined as `#define STR(X) #X` (rxm.h).

I added a short description and example in the comments in line rxm.h:66